extensions/file/README.md

```diff
@@ -0,0 +1,63 @@
+# File Info Extension Specification
+
+- **Title: File Info** 
+- **Identifier: file**
+- **Field Name Prefix: file**
+- **Scope: Item, Catalog, Collection**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+
+Provides a way to specify file related details such as checksum, data type and size for assets and links in [STAC Items](../../item-spec/item-spec.md), [STAC Catalogs](../../catalog-spec/catalog-spec.md) and [STAC Collections](../../collection-spec/collection-spec.md).
+
+- [Example](examples/sentinel1.json)
+- [JSON Schema](json-schema/schema.json)
+
+## *Link Object* and *Asset Object* fields
+
+The following fields can be used for Links (in the [`Link Object`](../../item-spec/item-spec.md#link-object)) and assets (in the [`Asset Object`](../../item-spec/item-spec.md#asset-object)).
+
+| Field Name         | Type   | Description                                                  |
+| ------------------ | ------ | ------------------------------------------------------------ |
+| file:byte_order | string | The byte order of integer values in the file. One of `big-endian` or `little-endian`. |
+| file:checksum | string | Provides a way to specify file [checksums](#checksums) (e.g. BLAKE2, MD5, SHA1, SHA2, SHA3). The hashes are self-identifying hashes as described in the [Multihash specification](https://github.com/multiformats/multihash) and must be encoded as hexadecimal (base 16) string with lowercase letters. |
+| file:data_type | string | The data type of the file. One of the [data types](#data-types) below. |
+| file:header_size | integer | The header [size](#sizes) of the file, specified in bytes. |
+| file:size | integer | The file [size](#sizes), specified in bytes. |
+
+This extension can OPTIONALLY be used with the [Collection Assets Extension](../collection-assets/README.md).
+File specific details should not be part of the [Item Assets Definition](../item-assets/README.md) in Collections.
+
+### Sizes
+
+Please be aware that the integer values (always unsigned) given for the sizes (especially `file:size`)  may exceed the maximum value for the default integer data type in your environment / programming language. In this specification `integer` specifies a integer number without an upper limit. You might need to use other data types to store the values in. For example, files with a size larger than around 2,14 GB would exceed the maximum value for int32 and in JavaScript `BigInt` could be used then.
+
+### Data Types
+
+- `int8`: 8-bit integer
+- `int16`: 16-bit integer
+- `int32`: 32-bit integer
+- `int64`: 64-bit integer
+- `uint8`: unsigned 8-bit integer (common for 8-bit RGB PNG's)
+- `unit16`: unsigned 16-bit integer
+- `uint32`: unsigned 32-bit integer
+- `uint64`: unsigned 64-bit integer
+- `float32`: 32-bit float
+- `float64`: 64-big float
+- `cint16`: 16-bit complex integer
+- `cint32`: 32-bit complex integer
+- `cfloat32`: 32-bit complex float
+- `cfloat64`: 64-bit complex float
```

Comment:
Some background on where these come from could be useful -- they're obvious reference points for rasterio / gdal users, but if I'm working with a frontend-only client and don't tend to work much with the underlying data I think these would look kind of out-of-nowhere.

Reply:
Took it from #923, not sure where people there took it from. I added (u)int64 though.

Reply:
Did you mean you took it from #879?

We had a discussion on gitter about what to do, generally its taken from rasterio/numpy. 

I agree we should give more background - I can try to take a crack at explaining it a bit more, but this is not my area of expertise, so will have to re-read the conversations to do so. And won't get to it super fast, so if anyone else wants to take a shot then go for it.

Reply:
Yes, indeed. I felt it might be a bit too restrictive so I added int64/uin64/float16, but still there might be additional types, like you find references to int128 online for example and also additions to complex ints/floats. But I'm also no expert in this...

Reply:
We could also go with `^(u|c)?(int|float)(8|16|32|64|128)$` with u = unsigned, c = complex and the number being the bits (but then there's likely no float8) ;-)

Anyway, I'm not sure what additional context you'd like to see here? There's likely no table that lists (standardizes) all of them. One could try to align with IEEE 754, but the names seem to be completely different than what is often used in tooling?!

Reply:
I think the context is just a couple lines on what these are used for - why a client might want this information, why providers should consider including it.

Reply:
I've added some context I could think of. Also added `other` for any files that may include other data types (e.g. larger number, string, boolean/bit). Could you check that again?